e2e/pvc.go

```diff
@@ -235,3 +236,43 @@ func checkPVSelectorValuesForPVC(f *framework.Framework, pvc *v1.PersistentVolum
 	}
 	return nil
 }
+
+func getMetricsForPVC(f *framework.Framework, pvc *v1.PersistentVolumeClaim, t int) error {
+	kubelet, err := getKubeletIP(f.ClientSet)
+	if err != nil {
+		return err
+	}
+
+	// kubelet needs to be started with --read-only-port=10255
+	cmd := fmt.Sprintf("curl --silent 'http://%s:10255/metrics'", kubelet)
```

Comment:
Nits:
1. Why not use 127.0.0.1? (so we don't need DNS, rely on network)?
2. '10255' is the default. Do we foresee it'll ever change? (https://github.com/Azure/AKS/issues/1601 ? https://github.com/kubernetes/kubeadm/issues/732 ?)

Reply:
1. 127.0.0.1 will be the container (Rook Ceph Toolbox), not an address where Kubelet listens on
2. port 10255 used to be the default, it is now disabled unless `--read-only-port` is passed to Kubelet (which we do with the `scripts/minikube.sh`, once that works)

As an alternative for (2), we could add additional permissions to some Pod where we can get the metrics from (the node may not be reachable from the system where the e2e tests got started, hence the toolbox pod). Not all pods have permission to read the non-readonly port, and that needs certificates to authenticate.

Reply:
it might be possible to get the metrics from the E2E framework, can we please check Is there any helper functions already available for this one at https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/metrics

Reply:
That looks like an option, I have added a commit that rewrites the test to use that.

Reply:
When using the metrics gathering from the Kubernetes E2E framework, I do get
```
kubelet_container_log_filesystem_used_bytes{container="rhel8", namespace="block-stats", pod="rhel8-rbd-block", uid="87d26b70-a1c4-4017-a29b-9336d9a187c5"} => 12288
```
This does not call the CSI NodeGetVolumeStats procedure, so it is not useful for this testing, unfortunately.